### PR TITLE
Do not crash napari viewer if error happen on open file via cli

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

Address second error from #6966

# Description

This Pr adds handling error of opening file using while passing file path via CLI

### CODE REVIEW

File: napari/__main__.py [hunk 1]
```diff
@@ -343,13 +344,18 @@ def _run() -> None:
...
+            viewer._window._qt_viewer._qt_open(
+                args.paths,
+                stack=args.stack,
+                plugin=args.plugin,
+                layer_type=args.layer_type,
+                **kwargs,
+            )
+        except ReaderPluginError:
+            logging.exception(
+                'Error happen during load file while starting napari'
```
Comment (jni) [inline napari/__main__.py]:
```suggestion
                'Loading %s with %s failed with errors', args.paths, args.plugin
```

Comment (psobolewskiPhD) [reply napari/__main__.py]: Should this be in a `trans`?

**Review by psobolewskiPhD — APPROVED** [review]
This is a good solution to the issue.
I concur with Juan's suggestion to improve the message, although here's what the traceback actually looks like:
```
22:06:55 : ERROR : MainThread : Error happen during load file while starting napari
Traceback (most recent call last):
  File "/Users/piotrsobolewski/Dev/napari/napari/components/viewer_model.py", line 1472, in _add_layer_from_data
    layer = add_method(data, **(meta or {}))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: add_labels() got an unexpected keyword argument 'color'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/piotrsobolewski/Dev/napari/napari/components/viewer_model.py", line 1281, in _open_or_raise_error
    added = self._add_layers_with_plugins(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/piotrsobolewski/Dev/napari/napari/components/viewer_model.py", line 1399, in _add_layers_with_plugins
    added.extend(self._add_layer_from_data(*_data))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/piotrsobolewski/Dev/napari/napari/components/viewer_model.py", line 1477, in _add_layer_from_data
    raise TypeError(
TypeError: _add_layer_from_data received an unexpected keyword argument ('color') for layer type labels

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/piotrsobolewski/Dev/napari/napari/_qt/qt_viewer.py", line 986, in _qt_open
    self.viewer.open(
  File "/Users/piotrsobolewski/Dev/napari/napari/components/viewer_model.py", line 1181, in open
    layers = self._open_or_raise_error(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/piotrsobolewski/Dev/napari/napari/components/viewer_model.py", line 1290, in _open_or_raise_error
    raise ReaderPluginError(
napari.errors.reader_errors.ReaderPluginError: Tried opening with napari-ome-zarr, but failed.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/piotrsobolewski/Dev/napari/napari/__main__.py", line 348, in _run
    viewer._window._qt_viewer._qt_open(
  File "/Users/piotrsobolewski/Dev/napari/napari/_qt/qt_viewer.py", line 994, in _qt_open
    handle_gui_reading(
  File "/Users/piotrsobolewski/Dev/napari/napari/_qt/dialogs/qt_reader_dialog.py", line 190, in handle_gui_reading
    readers = prepare_remaining_readers(paths, plugin_name, error)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/piotrsobolewski/Dev/napari/napari/_qt/dialogs/qt_reader_dialog.py", line 245, in prepare_remaining_readers
    raise ReaderPluginError(
napari.errors.reader_errors.ReaderPluginError: Tried to read https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.4/idr0001A/2551.zarr/A/1/0 with plugin napari-ome-zarr, because it was associated with that file extension/because it is the only plugin capable of reading that path, but it gave an error. Try associating a different plugin or installing a different plugin for this kind of file.
```
So the logging error from this is at the top and likely to be missed by the user, while the final exception is more visible and more or less the same as the suggestion.

File: napari/__main__.py (hunk 1 above)
Comment (jni) [reply napari/__main__.py]: imho logging doesn't need to be translated. At any rate, our current approach with translations is that we shouldn't worry too much about maintaining them, as they are currently unused, and the program will probably change *a lot* before we do need to use them, including turnover (wholesale deletion) of the messages. imho this means doing a best-guess while writing code, but not holding up reviews for it.

**Pull request merged**